app/lib/activitypub/activity/undo.rb

```diff
@@ -113,6 +115,22 @@ def undo_like
     end
   end
 
+  def undo_emoji_react
+    name = @object['content']
+    return if name.nil?
+
+    status = status_from_uri(target_uri)
+
+    return if status.nil? || !status.account.local?
+
+    if @account.reacted?(status, name.delete(':'))
+      reaction = status.status_reactions.where(account: @account, name: name).first
```

Comment:
reaction is nil when `name` is a custom emoji shortcode, so custom emoji reactions are never deleted.

Reply:
```suggestion
      reaction = status.status_reactions.where(account: @account, name: name.delete(':').first
```

*facepalm*